File: Hashi/src/main/java/fr/projetFormation/Hashi/entities/Restaurant.java
package fr.projetFormation.Hashi.entities;

import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

import javax.persistence.AttributeOverride;
import javax.persistence.AttributeOverrides;
import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "restaurant")
@SequenceGenerator(name = "seqRestaurant", sequenceName = "seq_restaurant", allocationSize = 1)
public class Restaurant {
	@Id
	@Column(name = "restaurant_id", nullable = false)
	@GeneratedValue(generator = "seqRestaurant", strategy = GenerationType.SEQUENCE)
	private Long id;
	@NotBlank
	@NotNull
	@Column(name = "restaurant_name")
	private String nom;
	@Embedded
	@AttributeOverrides({ @AttributeOverride(name = "numero", column = @Column(name = "restaurant_numero")),
			@AttributeOverride(name = "rue", column = @Column(name = "restaurant_rue")),
			@AttributeOverride(name = "codePostal", column = @Column(name = "restaurant_code_postal", length = 20)),
			@AttributeOverride(name = "ville", column = @Column(name = "restaurant_ville")) })
	private Adresse adresse;
	@OneToMany(mappedBy = "id.restaurant")
	private Set<LigneCarte> lignesCarte = new HashSet<LigneCarte>();

	@ManyToOne
	@JoinColumn(name = "restaurant_gerant_id", foreignKey = @ForeignKey(name = "restaurant_gerant_id_fk"))
	private Gerant gerant;
	@OneToMany(mappedBy = "restaurant")
	private Set<Cuisinier> cuisiniers = new HashSet<Cuisinier>();

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public Adresse getAdresse() {
		return adresse;
	}

	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}

	public Gerant getGerant() {
		return gerant;
	}

	public void setGerant(Gerant gerant) {
		this.gerant = gerant;
	}

	public Set<Cuisinier> getCuisiniers() {
		return cuisiniers;
	}

	public void setCuisiniers(Set<Cuisinier> cuisiniers) {
		this.cuisiniers = cuisiniers;
	}

	public Set<LigneCarte> getLignesCarte() {
		return lignesCarte;
	}

	public void setLignesCarte(Set<LigneCarte> lignesCarte) {
		this.lignesCarte = lignesCarte;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Restaurant other = (Restaurant) obj;
		return Objects.equals(id, other.id);
	}

	public Restaurant() {
	}

	public void rejetCommande(Commande commande) {
		// TODO
	}

	public void addCuisinier(Cuisinier cuisinier) {
		cuisiniers.add(cuisinier);
	}

	public void removeCuisinier(Cuisinier cuisinier) {
		cuisiniers.remove(cuisinier);
	}

}

File: HashiAngular/src/app/model/restaurant.ts
import { LigneCarte } from "./ligne-carte";
import { Adresse } from "src/app/model/adresse";
import { Gerant } from "./gerant";
import { Cuisinier } from "./cuisinier";
export class Restaurant {
  constructor(
    private _adresse?: Adresse | undefined,
    private _ligneCarte?: LigneCarte[] | undefined,
    private _id?: number | undefined,
    private _nom?: string | undefined,
    private _gerant?: Gerant | undefined,
    private _cuisiniers?: Cuisinier[] | undefined
    // private _codePostal?: string | undefined, // private _numero?: number | undefined, // private _rue?: string | undefined, // private _ville?: string | undefined
  ) {}

  /**
   * Getter id
   * @return {number}
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter nom
   * @return {string}
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Setter id
   * @param {number} value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter nom
   * @param {string} value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }
  /**
   * Getter adresse
   * @return {Adresse}
   */
  public get adresse(): Adresse | undefined {
    return this._adresse;
  }

  /**
   * Setter adresse
   * @param {Adresse} value
   */
  public set adresse(value: Adresse | undefined) {
    this._adresse = value;
  }
  /**
   * Getter ligneCarte
   * @return {LigneCarte}
   */
  public get ligneCarte(): LigneCarte[] | undefined {
    return this._ligneCarte;
  }

  /**
   * Setter ligneCarte
   * @param {LigneCarte} value
   */
  public set ligneCarte(value: LigneCarte[] | undefined) {
    this._ligneCarte = value;
  }

  public get gerant(): Gerant | undefined {
    return this._gerant;
  }

  public set gerant(value: Gerant | undefined) {
    this._gerant = value;
  }

  public get cuisiniers(): Cuisinier[] | undefined {
    return this._cuisiniers;
  }

  public set cuisiniers(value: Cuisinier[] | undefined) {
    this._cuisiniers = value;
  }

  public addCuisinier(value : Cuisinier){
    this.cuisiniers?.push(value);
  }
  // /**
  //  * Getter numero
  //  * @return {number}
  //  */
  // public get numero(): number | undefined {
  //   return this._numero;
  // }

  // /**
  //  * Getter rue
  //  * @return {string}
  //  */
  // public get rue(): string | undefined {
  //   return this._rue;
  // }

  // /**
  //  * Getter codePostal
  //  * @return {string}
  //  */
  // public get codePostal(): string | undefined {
  //   return this._codePostal;
  // }

  // /**
  //  * Getter ville
  //  * @return {string}
  //  */
  // public get ville(): string | undefined {
  //   return this._ville;
  // }

  // /**
  //  * Setter numero
  //  * @param {number} value
  //  */
  // public set numero(value: number | undefined) {
  //   this._numero = value;
  // }

  // /**
  //  * Setter rue
  //  * @param {string} value
  //  */
  // public set rue(value: string | undefined) {
  //   this._rue = value;
  // }

  // /**
  //  * Setter codePostal
  //  * @param {string} value
  //  */
  // public set codePostal(value: string | undefined) {
  //   this._codePostal = value;
  // }

  // /**
  //  * Setter ville
  //  * @param {string} value
  //  */
  // public set ville(value: string | undefined) {
  //   this._ville = value;
  // }
}
